doc/releases/changelog-dev.md

```diff
@@ -52,6 +52,10 @@
 * Added the `qml.TShift` and `qml.TClock` qutrit operations for qutrit devices, which are the qutrit analogs of the Pauli X and Pauli Z operations.
   ([#2841](https://github.com/PennyLaneAI/pennylane/pull/2841))
 
+* Added the private `_prod_sort` function that sorts a list of operators by their respective wires
+  taking into account their commutativity property.
+  [(#2995)](https://github.com/PennyLaneAI/pennylane/pull/2995)
+
```

Comment:
This isn't user-facing at all, so I'm not sure the changelog entry is necessary.

Reply:
The changelog's also for developers and logging purposes.  Though maybe at some point we should have an "Internal improvements" section for stuff like this.